YvesIraguha/lost-and-found-api
# #173527850 Integrating email notification for found item

### What does this PR do?

Sending email notification of found item to the owner

### Description of task to be completed?

- Integrating sendGrid to send email for the found item

- End point for the Integration of email notification for the found item

### How should this be manually tested?

- Clone the repo `git clone`

- Checkout the repo `git checkout ft-email-notification-#173527850 `

- Update your `.env` file with `.env.sample` data 

- Run  `npm install` to install all dependecies.

- Then run `npm run dev` to start the server.

- in postman send a post request on `http://localhost:3000/api/v1/items/lost` to register your lost item, data to send in the boy
```
{
     documentName: "documentName",
     documentNumber:"documentNumber"
     sector: "Sector",
     district: "District",
     isRewarded: "is rewarded",  //true or false
     price: "value" // price to be issued if is rewarded
}
```

- Then send another post request on `http://localhost:3000/api/v1/items/found` to re-register the registered lost item as found item.

- Then  check the email inbox of the user who registered item as lost, you will find message sent that document was found.

### Any background context you want to provide?

- For the url to work in order to receive email, when you are updating your `.env` file make sure to add `.trim()` after `SENDGRID_API_KEY` and assigned data will be found on slack

- In order to send a post request, user must login or signup if not have account and provide token generated to the header. By assigning key to `auth-token` and generated token in `value` of header.

- Price is required if isRewarded set to true

 ### What are the relevant pivotal tracker stories?

- #173527850
